### ISSUE TITLE
Set the montage outside the pipeline

### ISSUE THREAD

**Issue by yjmantilla** [NeuroDataDesign/pyprep#5]
I propose that the montage rather than the montage kind to be a parameter entering the PREP, Basically the only change would be in  PrepPipeline class:
```
def __init__(self, raw, prep_params, montage_kind="standard_1020", ransac=True,montageIn=None):
#code before
        #montage = mne.channels.read_montage(kind=montage_kind, ch_names=self.ch_names)
        if montageIn is None:
            montage = mne.channels.make_standard_montage(kind=montage_kind)
        else:
            montage = montageIn
        self.raw.set_montage(montage)
# code after
```
This is also helpful since read_montage is being deprecated https://github.com/mne-tools/mne-python/issues/6950

The change could also be implemented in the same way in the Noisydata class but it seems it is not being used anymore.

@adam2392 proposed that the montage be set up outside the pipeline, I agree with this:

> I somewhat agree! Although if that is the case, I wonder if it should've been attached to the raw beforehand using raw.set_montage(montage). If that is the case, really what is necessary is a Attribute_Exception if a montage is not set in the raw data structure passed in. Then have the example reflect that necessity.

Now, once one sets the montage basically what happens is that the raw.info.chs list of dictionaries is filled with the location info. This also happens with raw.info.dig list of dictionaries. I think the interpolation grabs the locations from chs dictionaries.

So basically we just have to make sure all the locations are available for the reference electrodes, this can be done with the following code:

```
    # Make sure that reference channels have locations for interpolation
    for channel in raw.info.get('chs'):
        if channel.get('loc') is None:
            raise Exception('performReference:NoChannelLocations,evaluation channels must have locations')
    print('performReference: Enough sensor locations')

    # Another way
    chanlocs = [channel.get('loc',None) for channel in raw.info.get('chs')]

``` 

Maybe there is an easier way though.
